src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/ConfigurableProductSubscriber.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ProductBundle\Form\EventSubscriber;
+
+use Sylius\Component\Product\Model\ProductInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+class ConfigurableProductSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        /** @var ProductInterface $product */
+        $product = $event->getData();
+
+        Assert::isInstanceOf($product, ProductInterface::class);
+
+        if ($product->isSimple()) {
+            return;
+        }
+
+        $form = $event->getForm();
+        $form->remove('masterVariant');
+
+        /** Options should be disabled for configurable product if it has at least one defined variant */
+        $disableOptions = (null !== $product->getMasterVariant());
+
+        $form->add('options', 'sylius_product_option_choice', [
+            'required' => false,
+            'disabled' => $disableOptions,
```

Comment:
Should be `'disabled' => $product->hasVariants(),`. Current logic will render not disabled options for configurable product with variants since this product will always return `null` on `getMasterVariant`